src/main/java/training/subsystems/Elbow/ElbowSubsystem.java

```diff
@@ -0,0 +1,76 @@
+package training.subsystems.Elbow;
+
+import com.revrobotics.CANSparkBase;
+import com.revrobotics.CANSparkLowLevel;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.SparkPIDController;
+import edu.wpi.first.math.geometry.Rotation2d;
+import utils.GBSubsystem;
+
+import static training.subsystems.Elbow.Constants.*;
+
+public class ElbowSubsystem extends GBSubsystem {
+    private static ElbowSubsystem instance;
+
+    private final CANSparkMax motor = new CANSparkMax(ELBOW_ID, CANSparkLowLevel.MotorType.kBrushless);
+    private final SparkPIDController motorPIDF = motor.getPIDController();
+    private Rotation2d angle = DEFAULT_POSITION_ELBOW;
+
+    private ElbowSubsystem() {
+        motorPIDF.setP(ELBOW_PID_CONTROLLER.getP());
+        motorPIDF.setD(ELBOW_PID_CONTROLLER.getD());
+        motorPIDF.setI(ELBOW_PID_CONTROLLER.getI());
+    }
+
+    public static ElbowSubsystem getInstance() {
+        if (instance == null) {
+            instance = new ElbowSubsystem();
+        }
+        return instance;
+    }
+
+    public void moveWritst(double power) {
+        if (Math.abs(power) < POWER_LIMIT_ELBOW) {
+            motor.set(power);
+        }
+    }
+
+    public Rotation2d getMotorAngle() {
+        return Rotation2d.fromRotations(motor.getEncoder().getPosition());
+    }
+
+    public Rotation2d getDesiredAngle() {
+        return angle;
+    }
+
+    public void setAngle(Rotation2d angle) {
+        this.angle = angle;
+    }
+
+    public void addToAngle(Rotation2d angle) {
+        this.angle = this.angle.plus(angle);
+    }
+
+    public void subtractFromAngle(Rotation2d angle) {
+        this.angle = this.angle.minus(angle);
+    }
+
+    public boolean isAtAngle(Rotation2d angle) {
+        double anglesDelta = (this.angle.getDegrees() % 1) - (angle.getRadians() % 1);
+        return Math.abs(anglesDelta) <= ELBOW_TOLERANCE;
+    }
+
+    @Override
+    protected String getLogPath() {
+        return ELBOW_LOGPATH;
+    }
+
+    @Override
+    public void subsystemPeriodic() {
+        motorPIDF.setReference(
+                angle.getRotations() / ELBOW_GEAR_RATIO % 1,
```

Comment:
why  / ELBOW_GEAR_RATIO % 1, also in what units is the refernce should be?

Reply:
according to the docs the units are rotations, which I used. I changed the order of operations to be `(angle.getRotations() % 1) / ELBOW_GEAR_RATIO`, which is supposed to make more sense (though it acts the same). I used `% 1` since I don't want the motor to do more than one rotation -- I care only about the "angle" (normalized to 0-1). Is there any other way to make it a bit more clear?

Reply:
if the motor is at 100 rotations and you tell him to go to 0.5, what will happen? u need more robust handling for this

Reply:
OK i have some weird thing instead, it should work